### ISSUE TITLE
no detections

### ISSUE THREAD

**Issue by VICTORYGS** [ultralytics/ultralytics#2470]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

my datasets:  https://public.roboflow.com/object-detection/mask-wearing/4/download/yolov8
![s](https://user-images.githubusercontent.com/33747619/236790635-d61880b2-a5dd-4cd3-b3c4-2cc35f09fcd9.jpg)

The model I trained has been unable to detect content. Does anyone know what's going on

**Comment by glenn-jocher** [comment]
Hi @VICTORYGS, thank you for reaching out about this issue. It's difficult to determine the problem without further information on how you trained the model or what the training process was like.

 However, some common reasons as to why a model trained with YOLOv8 may not be detecting objects include not providing enough data for the model to learn from, incorrect labeling of the data in terms of bounding boxes, and poor model performance due to bad hardware configurations.

We suggest you check to make sure your data labels are accurate and that your training data set is diverse enough to ensure good model performance. Let us know if this helps or if you have additional information that could help us give you a more accurate solution to your problem.

**Comment by sech-ai** [comment]
como puedo verificar que las etiquetas de datos son precisas ?

**Comment by glenn-jocher** [comment]
@sech-ai, ¡gracias por tu interés en YOLOv8! Para verificar la precisión de las etiquetas de datos, te sugeriría echar un vistazo más de cerca a tus datos de entrenamiento y etiquetas. Puedes seguir los siguientes pasos:

1. Primero, asegúrate de tener un conjunto diverso y representativo de imágenes que cubran todas las clases que deseas detectar. Tener suficientes ejemplos en cada clase puede mejorar el rendimiento de detección.

2. Revisa manualmente algunas muestras de tus imágenes de entrenamiento y asegúrate de que las etiquetas de las cajas delimitadoras sean precisas y estén correctamente alineadas con los objetos que deseas detectar. Esto puede incluir verificar que las cajas delimitadoras abarquen todo el objeto de interés y que no haya superposición o solapamiento incorrecto entre objetos.

3. Si tienes dudas sobre las etiquetas, puedes realizar una revisión manual de tu conjunto de datos para verificar la precisión en un subconjunto más pequeño. Esto puede ser especialmente útil si observas resultados inconsistentes o incorrectos en las detecciones.

Recuerda que siempre es importante tener un conjunto de datos de entrenamiento de calidad y etiquetas precisas para obtener buenos resultados. Si continúas teniendo dificultades con la detección, ¡no dudes en proporcionar más información y estaremos encantados de ayudarte más!

**Comment by JB-Rockstar** [comment]
I got the 'no detection' text with my custom dataset during prediction. There was no detection for any of the prediction set files. Also, the graphs in the train folder were deficient.

In the train code, instead of `model = YOLO("yolov8s.yaml")` , I tried `model = YOLO("yolov8s.pt")`. It worked! The predict code did a good job of detecting and the graphs have improved.

**Comment by glenn-jocher** [comment]
@JB-Rockstar hi there, 

It sounds like you were able to resolve your "no detection" issue by loading a pre-trained YOLOv8 model (.pt file) instead of initializing a new model from the configuration file (.yaml file). 

When you initialize your YOLOv8 with "yolov8s.yaml", you're likely starting the training from scratch with randomly initialized weights. Without pre-training, the model might require more data or more epochs to learn effectively, which could have led to no detections.

On the other hand, when you load the model from "yolov8s.pt", you are most likely using a pre-trained model. This means that the model has already learned useful features from a large dataset, which can help it make good predictions, even with less data or fewer epochs. 

I'm glad to hear that your issue has been resolved! If you have any more questions or face any other issues, please feel free to bring them up. Happy training!

**Comment by NeoHe668** [comment]
> I got the 'no detection' text with my custom dataset during prediction. There was no detection for any of the prediction set files. Also, the graphs in the train folder were deficient.
> 
> In the train code, instead of `model = YOLO("yolov8s.yaml")` , I tried `model = YOLO("yolov8s.pt")`. It worked! The predict code did a good job of detecting and the graphs have improved.

I've got the same issue, it works, tks

**Comment by pderrenger** [comment]
Great to hear that switching to the pre-trained model (`yolov8s.pt`) resolved the issue for you as well! Pre-trained models often provide a better starting point, leveraging learned features from large datasets. If you encounter any further issues or have additional questions, feel free to ask.

**Comment by jacksonthall22** [comment]
I'm getting this issue too. I am positive my data is labelled correctly (it is output directly from `labelImg` into YOLO format - `(x_center, y_center, w, h)` all as relative values between 0 and 1). I'm using a pretrained model (`yolov8n.pt`) and I trained for 50 epochs, where loss appears to go down. I have a small dataset of 20 labeled images, but I trained for 50 epochs, so I expected at least to get **some** bounding boxes outputted (otherwise how is the `box_loss` going down?). Here are some results from the training:

`results.csv`
```csv
                  epoch,         train/box_loss,         train/cls_loss,         train/dfl_loss,   metrics/precision(B),      metrics/recall(B),       metrics/mAP50(B),    metrics/mAP50-95(B),           val/box_loss,           val/cls_loss,           val/dfl_loss,                 lr/pg0,                 lr/pg1,                 lr/pg2
                      1,                  1.541,                 3.5546,                 1.7733,                0.01944,                   0.55,                0.13049,                0.04193,                   1.34,                  3.564,                 2.1066,                      0,                      0,                      0
                      2,                 1.5266,                 3.5657,                 1.7277,                0.02068,                  0.575,                0.15448,                0.04586,                 1.3387,                 3.5585,                  2.083,              1.089e-05,              1.089e-05,              1.089e-05
                      3,                 1.5357,                 3.5499,                  1.756,                0.02261,                    0.6,                0.14319,                0.05105,                  1.337,                 3.5593,                 2.0152,              2.134e-05,              2.134e-05,              2.134e-05
                      4,                 1.5136,                 3.5562,                 1.7335,                0.02666,                    0.7,                0.14428,                0.05261,                  1.288,                 3.5509,                  1.939,              3.135e-05,              3.135e-05,              3.135e-05
                      5,                 1.2839,                 3.5158,                 1.5876,                0.02438,                  0.675,                0.18152,                0.06135,                 1.2273,                 3.5333,                 1.8381,              4.092e-05,              4.092e-05,              4.092e-05
                      6,                 1.1683,                 3.4523,                 1.5059,                0.02858,                  0.775,                0.24027,                0.08385,                 1.1893,                 3.4911,                 1.7209,             5.0051e-05,             5.0051e-05,             5.0051e-05
                      7,                  1.034,                 3.3461,                 1.3784,                0.03006,                    0.8,                0.31547,                0.13867,                 1.1298,                 3.4246,                  1.571,             5.8741e-05,             5.8741e-05,             5.8741e-05
                      8,                0.93348,                 3.3382,                 1.3193,                0.02811,                    0.8,                0.38699,                0.17698,                 1.0603,                 3.3594,                  1.415,             6.6991e-05,             6.6991e-05,             6.6991e-05
                      9,                0.93307,                 3.2654,                 1.3136,                0.02696,                    0.8,                0.41831,                0.23404,                0.87129,                 3.2865,                 1.2389,             7.4801e-05,             7.4801e-05,             7.4801e-05
                     10,                 0.8179,                 3.1654,                 1.2166,                0.02653,                    0.8,                0.44529,                0.30347,                0.80056,                 3.2506,                 1.1703,             8.2172e-05,             8.2172e-05,             8.2172e-05
                     11,                 0.7971,                 3.1122,                 1.2044,                 0.0268,                    0.8,                0.47794,                0.35908,                0.74417,                 3.2326,                 1.1352,             8.9102e-05,             8.9102e-05,             8.9102e-05
                     12,                0.78863,                 3.1058,                 1.1689,                0.02749,                   0.85,                0.52438,                0.39313,                0.68121,                 3.2113,                  1.102,             9.5593e-05,             9.5593e-05,             9.5593e-05
                     13,                0.76122,                 2.9538,                 1.1889,                0.02792,                   0.85,                0.55528,                0.45648,                0.63579,                  3.198,                 1.0661,             0.00010164,             0.00010164,             0.00010164
                     14,                0.68417,                 2.9377,                 1.1106,                0.02878,                   0.85,                0.58351,                0.50904,                0.57888,                 3.1828,                 1.0381,             0.00010725,             0.00010725,             0.00010725
                     15,                0.67554,                 2.7814,                 1.1078,                0.03008,                    0.9,                0.60891,                  0.539,                0.54123,                 3.1558,                 1.0158,             0.00011242,             0.00011242,             0.00011242
                     16,                 0.6347,                 2.6625,                 1.0661,                0.03205,                    0.9,                0.60882,                0.52047,                0.55421,                 3.1456,                 1.0045,             0.00011715,             0.00011715,             0.00011715
                     17,                0.59781,                 2.4896,                 1.0655,                0.03261,                   0.85,                0.60944,                0.51934,                0.54951,                 3.1188,                0.99562,             0.00012145,             0.00012145,             0.00012145
                     18,                0.64294,                  2.385,                 1.0707,                0.03261,                   0.85,                0.60944,                0.51934,                0.54951,                 3.1188,                0.99562,              0.0001253,              0.0001253,              0.0001253
                     19,                0.64739,                 2.4984,                 1.0807,                0.03365,                   0.85,                0.60862,                0.51628,                0.55494,                 3.0692,                 0.9866,             0.00012871,             0.00012871,             0.00012871
                     20,                0.66231,                 2.2784,                  1.104,                0.03365,                   0.85,                0.60862,                0.51628,                0.55494,                 3.0692,                 0.9866,             0.00013168,             0.00013168,             0.00013168
                     21,                0.58645,                 2.2827,                 1.0566,                0.03634,                   0.85,                0.65137,                0.55446,                0.55509,                 3.0116,                0.97201,             0.00013421,             0.00013421,             0.00013421
                     22,                0.59573,                 2.1495,                 1.0636,                0.03634,                   0.85,                0.65137,                0.55446,                0.55509,                 3.0116,                0.97201,              0.0001363,              0.0001363,              0.0001363
                     23,                0.58444,                 2.1638,                 1.0642,                 0.0367,                   0.85,                0.70736,                0.61543,                0.53634,                 2.9401,                0.95634,             0.00013795,             0.00013795,             0.00013795
                     24,                0.63042,                 2.0718,                 1.0946,                 0.0367,                   0.85,                0.70736,                0.61543,                0.53634,                 2.9401,                0.95634,             0.00013916,             0.00013916,             0.00013916
                     25,                0.58649,                 2.0181,                 1.0671,                0.03924,                   0.85,                0.69218,                0.60846,                0.52042,                 2.8601,                0.94268,             0.00013993,             0.00013993,             0.00013993
                     26,                 0.6064,                 1.8891,                 1.1087,                0.03924,                   0.85,                0.69218,                0.60846,                0.52042,                 2.8601,                0.94268,             0.00014026,             0.00014026,             0.00014026
                     27,                0.56777,                  1.954,                 1.0599,                0.04183,                   0.85,                0.66408,                0.59196,                0.50729,                 2.7789,                0.93335,             0.00014015,             0.00014015,             0.00014015
                     28,                0.62419,                 1.8504,                 1.0842,                0.04183,                   0.85,                0.66408,                0.59196,                0.50729,                 2.7789,                0.93335,             0.00013961,             0.00013961,             0.00013961
                     29,                0.55358,                 1.8491,                 1.0707,                0.04567,                   0.85,                0.66404,                0.59252,                0.49988,                  2.703,                 0.9316,             0.00013862,             0.00013862,             0.00013862
                     30,                0.53202,                 1.7652,                 1.0495,                0.04567,                   0.85,                0.66404,                0.59252,                0.49988,                  2.703,                 0.9316,             0.00013719,             0.00013719,             0.00013719
                     31,                0.50764,                 1.6839,                 1.0037,                0.05137,                   0.85,                0.65066,                0.59017,                0.47958,                 2.6372,                0.92549,             0.00013532,             0.00013532,             0.00013532
                     32,                0.51574,                  1.645,                 1.0376,                0.05137,                   0.85,                0.65066,                0.59017,                0.47958,                 2.6372,                0.92549,             0.00013301,             0.00013301,             0.00013301
                     33,                 0.5418,                 1.6633,                 1.0401,                 0.6877,                    0.6,                0.63541,                0.57906,                0.47644,                 2.5793,                0.93084,             0.00013026,             0.00013026,             0.00013026
                     34,                0.55143,                 1.6023,                 1.0333,                 0.6877,                    0.6,                0.63541,                0.57906,                0.47644,                 2.5793,                0.93084,             0.00012707,             0.00012707,             0.00012707
                     35,                 0.5391,                 1.5805,                 1.0229,                0.95924,                0.54253,                0.62398,                0.57771,                0.48997,                 2.5301,                0.93802,             0.00012345,             0.00012345,             0.00012345
                     36,                0.53316,                 1.5909,                 1.0252,                0.95924,                0.54253,                0.62398,                0.57771,                0.48997,                 2.5301,                0.93802,             0.00011938,             0.00011938,             0.00011938
                     37,                0.55497,                 1.4989,                 1.0369,                0.97297,                0.47134,                0.61794,                0.56331,                0.48813,                 2.4923,                0.94086,             0.00011487,             0.00011487,             0.00011487
                     38,                0.52926,                 1.4742,                 1.0117,                0.97297,                0.47134,                0.61794,                0.56331,                0.48813,                 2.4923,                0.94086,             0.00010992,             0.00010992,             0.00010992
                     39,                0.52566,                 1.5084,                 1.0336,                      1,                0.31588,                0.62454,                0.57037,                0.49025,                 2.4583,                0.94276,             0.00010453,             0.00010453,             0.00010453
                     40,                0.49462,                 1.4063,                0.97946,                      1,                0.31588,                0.62454,                0.57037,                0.49025,                 2.4583,                0.94276,             9.8703e-05,             9.8703e-05,             9.8703e-05
                     41,                0.49357,                 1.0822,                0.95889,                0.97404,                    0.4,                0.63012,                0.57044,                0.49619,                  2.422,                0.94251,             9.2435e-05,             9.2435e-05,             9.2435e-05
                     42,                0.51033,                 1.1097,                 1.0072,                0.97404,                    0.4,                0.63012,                0.57044,                0.49619,                  2.422,                0.94251,             8.5727e-05,             8.5727e-05,             8.5727e-05
                     43,                0.41844,                 1.0079,                0.93317,                0.96824,                0.45938,                 0.6376,                0.57291,                0.49393,                 2.3795,                0.93722,             7.8579e-05,             7.8579e-05,             7.8579e-05
                     44,                0.41261,                0.97283,                0.92845,                0.96824,                0.45938,                 0.6376,                0.57291,                0.49393,                 2.3795,                0.93722,             7.0991e-05,             7.0991e-05,             7.0991e-05
                     45,                0.42598,                0.98332,                 0.9409,                0.96441,                0.50682,                0.63865,                0.57669,                0.48171,                 2.3338,                0.93031,             6.2963e-05,             6.2963e-05,             6.2963e-05
                     46,                 0.5182,                 1.0508,                 1.0404,                0.96441,                0.50682,                0.63865,                0.57669,                0.48171,                 2.3338,                0.93031,             5.4495e-05,             5.4495e-05,             5.4495e-05
                     47,                0.39315,                0.93426,                0.88817,                0.96222,                0.50843,                0.64015,                0.58262,                0.47292,                 2.2829,                0.92554,             4.5587e-05,             4.5587e-05,             4.5587e-05
                     48,                0.42848,                0.93331,                0.92593,                0.96222,                0.50843,                0.64015,                0.58262,                0.47292,                 2.2829,                0.92554,             3.6239e-05,             3.6239e-05,             3.6239e-05
                     49,                0.43393,                 1.0091,                0.98255,                0.96019,                0.52667,                0.64194,                0.58021,                0.47091,                 2.2307,                0.92503,             2.6451e-05,             2.6451e-05,             2.6451e-05
                     50,                0.43001,                0.87912,                0.91902,                0.96019,                0.52667,                0.64194,                0.58021,                0.47091,                 2.2307,                0.92503,             1.6223e-05,             1.6223e-05,             1.6223e-05
```

![image](https://github.com/user-attachments/assets/7064648e-0c63-4c34-a418-729e37c5d3bb)

Any insight? Do I really just need more data?

**Comment by pderrenger** [comment]
It looks like your training process is progressing, as indicated by the decreasing loss values. However, the small dataset size (20 images) is likely insufficient for the model to generalize well, leading to poor detection performance. Increasing the dataset size should improve results. Additionally, ensure that your validation set is representative of the training data. If the issue persists, try using data augmentation techniques to artificially increase the dataset size and variability. If you haven't already, updating to the latest version of the Ultralytics package might also help resolve any underlying issues.

**Comment by jacksonthall22** [comment]
I'm just confused. My question remains, how would the bbox loss be going down if the model outputs nothing at all - no bounding boxes - on any of the train or test images?

Also,

> Let us know if this helps or if you have additional information that could help us give you a more accurate solution to your problem.
> I'm glad to hear that your issue has been resolved! If you have any more questions or face any other issues, please feel free to bring them up. Happy training!
> If you encounter any further issues or have additional questions, feel free to ask.

no offense, but your chatgpt is showing. I hope you guys at least verify the output? Lol

**Comment by pderrenger** [comment]
I understand your confusion. The bbox loss can decrease during training even if no bounding boxes are detected because the model might be learning to minimize the loss function without making correct predictions. This can happen if the dataset is too small or not representative enough. Increasing the dataset size and ensuring diverse and accurate annotations can help. Additionally, make sure you're using the latest version of the Ultralytics package to avoid any potential bugs. If the issue persists, consider experimenting with different hyperparameters or data augmentation techniques to improve model performance.
